### PR TITLE
Filter Functionality

### PR DESCRIPTION
**Pull request by ark2002**
- Added working product sorting and filtering functionality .

### CODE REVIEW

File: src/frontend/screens/ProductsScreen/ProductsScreen.js [hunk 1]
```diff
@@ -1,11 +1,24 @@
...
 
 function ProductsScreen() {
+  const { products } = useProducts();
+  const { filters: { sortBy, categories, rating, maxPrice }, dispatchFilters } = useFilters();
+
+  const sortedProducts = ProductsSorted(products, sortBy);
+  const categorizedProducts = ProductFilterCategory(sortedProducts, categories);
+  const ratedProducts = ProductFilterRated(categorizedProducts, rating);
+  const rangedProducts = ProductsMaxPrice(ratedProducts, maxPrice)
+  console.log("max price :" + maxPrice);
```
Comment (MarkVed17) [inline src/frontend/screens/ProductsScreen/ProductsScreen.js:15]: You may remove this.

Comment (ark2002) [reply src/frontend/screens/ProductsScreen/ProductsScreen.js:15]: yup ill take care of it in the next PR

**Review by vijaytembugade — APPROVED** [review]
Looks good to me. Nice work buddy

**Review by yogesh- — APPROVED** [review]
Please add deploy preview for future pr
- File structure is good
- code looks cleaner
- smaller components

**Pull request merged**